packages/framework/src/Component/UploadedFile/UploadedFileFacade.php

```diff
@@ -98,25 +101,26 @@ public function manageFiles(object $entity, UploadedFileData $uploadedFileData,
      * @param object $entity
      * @param string $entityName
      * @param string $type
-     * @param array $temporaryFilenames
+     * @param string $temporaryFilename
+     * @param string $uploadedFileName
      */
-    protected function uploadFile(object $entity, string $entityName, string $type, array $temporaryFilenames): void
+    protected function uploadFile(object $entity, string $entityName, string $type, string $temporaryFilename, string $uploadedFileName): void
     {
-        if (count($temporaryFilenames) > 0) {
-            $entityId = $this->getEntityId($entity);
+        $entityId = $this->getEntityId($entity);
+
+        $this->deleteAllUploadedFilesByEntity($entity);
 
-            $this->deleteAllUploadedFilesByEntity($entity);
+        $newUploadedFile = $this->uploadedFileFactory->create(
+            $entityName,
+            $entityId,
+            $type,
+            $temporaryFilename
+        );
 
-            $newUploadedFile = $this->uploadedFileFactory->create(
-                $entityName,
-                $entityId,
-                $type,
-                array_pop($temporaryFilenames)
-            );
+        $newUploadedFile->setNameAndSlug($uploadedFileName);
```

Comment:
Why is not this set in `UploadFile::__construct()`?

Reply:
Fixed